### PR TITLE
Use refresh=False for tqdm

### PR DESCRIPTION
**Pull request by fehiepsi**
This PR uses `refresh=False` for progressbar. The hanging issue does not only happen in my system but also a user in forum.

#### Tested:
+ one chain + terminal/notebook + CPU/GPU + old/new api
+ multi chain + CPU/GPU + terminal + old api
+ multi chain + CPU + notebook + old api

#### What is not tested:
+ multi chain + GPU + notebook: It is a bit tricky to run multi chains in GPU in notebook. New version of pytorch only supports `spawn`/`forkserver` (see [the docs](https://pytorch.org/docs/master/notes/multiprocessing.html#cuda-in-multiprocessing)). On the other hand, [it is tricky to use spawn method](https://github.com/pytorch/pytorch/issues/20375) in jupyter notebook. MCMC does not run when I use those methods with CUDA tensors in notebooks.

Anyway, the above issue does not related to this PR because progress bars work well in notebook with CPU tensors.

### CODE REVIEW

**Comment by neerajprad** [comment]
Just to confirm - multi-chain with the new API as well works for both notebook (CPU) and terminal?

**Comment by fehiepsi** [comment]
[tqdm docs](https://tqdm.github.io/docs/tqdm/) said that "if your progress is erratic with both fast and slow iterations (network, skipping items, etc) you should set miniters=1". This problem is [observed in tqdm](https://github.com/tqdm/tqdm/issues/249) with the solution is to tweak `miniters` and `mininterval`. I have played with those settings. They help somehow but the problem still happens.

Here is an example script which hangs when running in notebook
```
import torch
import pyro
import pyro.distributions as dist
from pyro.infer.mcmc.api import MCMC, NUTS

def model(sigma, y):
    eta = pyro.sample('eta', dist.Normal(torch.zeros(J), torch.ones(J)))
    mu = pyro.sample('mu', dist.Normal(torch.zeros(1), 10 * torch.ones(1)))
    tau = pyro.sample('tau', dist.HalfCauchy(scale=25 * torch.ones(1)))
    theta = mu + tau * eta
    return pyro.sample("obs", dist.Normal(theta, sigma), obs=y)

J = 8
y = torch.tensor([28,  8, -3,  7, -1,  1, 18, 12]).type(torch.Tensor)
sigma = torch.tensor([15, 10, 16, 11, 9, 11, 10, 18]).type(torch.Tensor)
pyro.set_rng_seed(0)
nuts_kernel = NUTS(model, jit_compile=True)
mcmc = MCMC(nuts_kernel, num_samples=1000, warmup_steps=1000, num_chains=1)
posterior = mcmc.run(sigma, y)
```

**Comment by fehiepsi** [comment]
> Just to confirm - multi-chain with the new API as well works for both notebook (CPU) and terminal?

I just test the above example with new API. multi-chain in notebook (CPU) works well. So I think it will work well in terminal too.

**Comment by neerajprad** [comment]
> Here is an example script which hangs when running in notebook

This still seems to work fine for me both for `num_chains = 1, 2`. My guess is that this is highly dependent on platform / random seed etc. What is the version of tqdm that you are using?

The issue that you linked relates to how frequently the progress bar should be updated. `tqdm` uses `miniters` and `mininterval` to decide this and when this is set dynamically (which is the default), the first few iterations will be used to learn this value. However, if the first few iterations are really slow but the remaining iterations are really fast, then the progress bar might not be updated until much later and the program may have to wait for this update if the loop has already terminated. With `refresh=True` we are practically forcing a refresh per sample, i.e. setting `miniter=1`, so this should actually fix such freezes. If setting `refresh=False` fixes the freezing issue instead, it probably points to another source of bug. 😕

**Comment by fehiepsi** [comment]
For an independent of Pyro example, non-random repro script, could you take a look at
```
import time
import tqdm
from collections import OrderedDict

def f(n):
    if n % 20 == 1 or n % 20 == 3 or n % 20 == 10:
        time.sleep(0.1)
        return "slow"
    else:
        time.sleep(0.01)
        return "fast"

t = tqdm.tqdm(total=2000)

for i in range(2000):
    postfix = OrderedDict(stat=f(i))
    t.set_postfix(postfix, refresh=True)
    t.update()
t.close()
```

This will hang my notebook after running weirdly from iteration 300. There is no way to stop it unless I use `kill -9 [pid]`. I use `tqdm=4.32.2`. Maybe this only happens in my linux system or a few other platforms. I don't think that `refresh=True` plays any role here when we already have `update()`, which will update the progress bar regardless whatever info we have updated.

I agree that this is a expected/unexpected tqdm issue. But rather than following up with this issue and waiting for a (possibly) fix in future version of tqdm, we can use `refresh=False`. At least it has no harm because we have `update()`.

**Comment by neerajprad** [comment]
I tried using versions `4.32.1` and `4.32.2` and they both ran fine. It is probably a platform specific issue. 

> But rather than following up with this issue and waiting for a (possibly) fix in future version of tqdm, we can use refresh=False. 

Agreed, I just wanted to point out that this is likely some bug in tqdm that should be fixed upstream. I'm happy to merge this fix once the build passes!

**Comment by neerajprad** [comment]
I'm not sure why this has triggered two travis builds, and one of them has been queued for quite a while now.
